lib/mysqlQuery.js

```diff
@@ -19,23 +19,38 @@ class MysqlQuery {
         }
     }
 
+    static validateUpsert(query) {
+        if (!MysqlQuery.isUpsert(query)) {
+            throw new Error("Invalid INSERT ... ON DUPLICATE KEY UPDATE query");
+        }
+    }
+
     static validateDelete(query) {
         if (!MysqlQuery.isDelete(query)) {
             throw new Error("Invalid DELETE query");
         }
     }
 
     static isSelect(query) {return MysqlQuery._isType(query, 'select');}
-    static isInsert(query) {return MysqlQuery._isType(query, 'insert');}
+    static isInsert(query) {return MysqlQuery._isType(query, 'insert') && !MysqlQuery._contains(query, 'on duplicate key update');}
```

Comment:
This breaks backwards compatibility so I wouldn't add this check.  I know for example this would bust the code I wrote at Foundry :)